crates/router/src/connector/expresscheckout/transformers.rs

```diff
@@ -0,0 +1,363 @@
+                    save_to_locker: false,
+                })
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        }
+    }
+}
+
+pub struct ExpresscheckoutAuthType {
+    pub(super) api_key: String,
+}
+
+impl TryFrom<&types::ConnectorAuthType> for ExpresscheckoutAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        if let types::ConnectorAuthType::HeaderKey { api_key } = auth_type {
+            Ok(Self {
+                api_key: api_key.to_string(),
+            })
+        } else {
+            Err(errors::ConnectorError::FailedToObtainAuthType)?
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum ExpresscheckoutPaymentStatus {
+    Charged,
+    Success,
+    Created,
+    PendingVbv,
+    AuthenticationFailed,
+    #[default]
+    AuthorizationFailed,
+    JuspayDeclined,
+    CaptureInitiated,
+    CaptureFailed,
+    VoidInitiated,
+    VoidFailed,
+    Voided,
+    Authorized,
+    Authorizing,
+    PendingAuthentication,
+}
+
+impl From<ExpresscheckoutPaymentStatus> for enums::AttemptStatus {
+    fn from(item: ExpresscheckoutPaymentStatus) -> Self {
+        match item {
+            ExpresscheckoutPaymentStatus::Charged => Self::Charged,
+            ExpresscheckoutPaymentStatus::Success => Self::Charged,
+            ExpresscheckoutPaymentStatus::Created => Self::Started,
+            ExpresscheckoutPaymentStatus::AuthenticationFailed => Self::AuthenticationFailed,
+            ExpresscheckoutPaymentStatus::AuthorizationFailed => Self::AuthorizationFailed,
+            ExpresscheckoutPaymentStatus::Authorizing => Self::Authorizing,
+            ExpresscheckoutPaymentStatus::Authorized => Self::Authorized,
+            ExpresscheckoutPaymentStatus::PendingVbv => Self::AuthenticationPending,
+            ExpresscheckoutPaymentStatus::JuspayDeclined => Self::Failure,
+            ExpresscheckoutPaymentStatus::CaptureFailed => Self::CaptureFailed,
+            ExpresscheckoutPaymentStatus::CaptureInitiated => Self::CaptureInitiated,
+            ExpresscheckoutPaymentStatus::VoidInitiated => Self::VoidInitiated,
+            ExpresscheckoutPaymentStatus::VoidFailed => Self::VoidFailed,
+            ExpresscheckoutPaymentStatus::Voided => Self::Voided,
+            ExpresscheckoutPaymentStatus::PendingAuthentication => Self::AuthenticationPending,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct ExpresscheckoutPaymentsResponse {
+    #[serde(default, deserialize_with = "deserialize_error_default")]
+    status: ExpresscheckoutPaymentStatus,
+    txn_uuid: Option<String>,
+    payment: Option<Authentication>,
+}
+
+fn deserialize_error_default<'de, D, ExpresscheckoutPaymentStatus>(
```

Comment:
I understand that this logic is to cover the scenario where we get unknown status from the connector. In such cases this deserializer will convert the status to default which is `AuthorizationFailed ` 

Is this a valid logic where we convert unknown status  into `AuthorizationFailed,` The reason I am raising this is `ExpresscheckoutPaymentsResponse ` is being used in Authorise, Capture, Cancel flows, So in any of these flows if we receive unknown status then by default `AuthorizationFailed ` will be sent which is wrong. Let's not have a custom deserialisation logic, if we receive any unknown status we will get deserialise error let it be.